### DISCUSSION TITLE
Google's oauth needing mitigation

### DISCUSSION THREAD

**Discussion by dabido3** [teslamate-org/teslamate#2685] [Q&A]
email received from google pertaining to my teslamate auto-backup flow:
"Our records indicate you have OAuth clients that used the OAuth OOB flow in the past." "Starting October 3, 2022, we will block OOB requests to Google’s OAuth 2.0 authorization endpoint for existing clients."

"Migrate your app(s) to an appropriate alternative method by following these instructions:
Determine your app(s) client type from your Google Cloud project by following the client links below.
Migrate your app(s) to a more secure alternative method by following the instructions in the blog post above for your client type."

This is a bit above my head -- could anyone offer some explanation of what needs to be done to mitigate the block implementing in Oct?

Do I need to change something on the teslamate side AND the google side? Unfortunately unsure where to begin.

Thanks in advance.

**Comment by tobiasehlert** [comment]
Hi @dabido3,

This sounds to me as something that is not related to TeslaMate itself.
Are you using some additional software that you use to authenticate with before you access your TeslaMate installation?

Regards,
Tobias

**Reply by dabido3** [reply]
Apologies for the muddy clarity…this is outside of the original Teslamate install but had hoped others might be in the same situation. this is referring to an automatic rclone cloud back up of the Teslamate data that uploads to Google drive every night. 
I had set this up about two years ago via instructions from 
https://www.teslaev.co.uk/how-to-perform-an-automatic-teslamate-backup-to-google-drive/

**Reply by cwanja** [reply]
Here is a TMC thread about it, I recall seeing some other users post about it. https://teslamotorsclub.com/tmc/threads/teslamate-backup-google-are-going-to-block-oob-requests-to-google’s-oauth-2-0-authorization-endpoint.265959/

It does not seem like there is a great, "simple" solution to this. I would suspect there might be some guides popup as October gets closer.
